### PR TITLE
Add warning for overlapping highlights

### PR DESCRIPTION
**Pull request by RoyEJohnson**
[DISCO-49]

[DISCO-49]: https://openstax.atlassian.net/browse/DISCO-49?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Dantemss — APPROVED** [review]
During testing make sure the warning doesn't trigger if you make the text selection empty or anything weird like that

File: src/app/messages/es/messages.json [hunk 1]
```diff
@@ -125,6 +125,7 @@
   "i18n:notification:toast:highlights:popup:load-failure": "No se pudo cargar algunos textos resaltados. Por favor, compruebe su conexión y vuelva a intentarlo.",
   "i18n:notification:toast:highlights:popup:print-failure": "No podemos imprimir esta página. Por favor, compruebe su conexión y vuelva a intentarlo.",
   "i18n:notification:toast:study-guides:load-failure": "No podemos cargar las guías de estudio en esta página. Por favor, compruebe su conexión y vuelva a intentarlo.",
+  "i18n:notification:toast:highlights:select-overlap": "No se admiten resaltados superpuestos.",
```
Comment (RoyEJohnson) [inline src/app/messages/es/messages.json:128]: @bethshook Can you check this translation, please?

File: src/app/messages/pl/messages.json [hunk 2]
```diff
@@ -125,6 +125,7 @@
   "i18n:notification:toast:highlights:popup:load-failure": "Niektóre zakreślenia na tej stronie nie mogą zostać załadowane. Sprawdź połączenie z internetem i spróbuj ponownie.",
   "i18n:notification:toast:highlights:popup:print-failure": "Nie możemy wydrukować tej strony. Sprawdź połączenie z internetem i spróbuj ponownie.",
   "i18n:notification:toast:study-guides:load-failure": "Nie możemy załadować zawartości Kompendium na tej stronie. Sprawdź połączenie z internetem i spróbuj ponownie.",
+  "i18n:notification:toast:highlights:select-overlap": "Nakładające się podświetlenia nie są obsługiwane.",
```
Comment (RoyEJohnson) [inline src/app/messages/pl/messages.json:128]: @paulinaszczucinska Can you check this translation of "Overlapping highlights are not supported" ?

File: src/app/messages/es/messages.json (hunk 1 above)
Comment (bethshook) [reply src/app/messages/es/messages.json:128]: looks good

File: src/app/messages/pl/messages.json (hunk 2 above)
Comment (paulinaszczucinska) [reply src/app/messages/pl/messages.json:128]: Sure @RoyEJohnson, it would be: "Nakładające się zaznaczenia nie są obsługiwane."

**Comment by Malar-Natarajan** [comment]
@RoyEJohnson The unit tests are failing in this PR. Could you please have a look

**Pull request merged**